### PR TITLE
widget initialization

### PR DESCRIPTION
**Pull request by alexgul1**
Added:

- widget initializition with appId and querySelector/DOMElement
- requests for getting app configuration and auth link configuration
- catch response errors

### CODE REVIEW

File: src/AuthPM.js [hunk 1]
```diff
@@ -1,6 +1,75 @@
...
+    btn.addEventListener('click', clickHandler);
+
+    return btn;
+  }
+
+  static #openLoginWindow(urlConfig) {
+    console.log(urlConfig);
+  }
+
+  static #isDomElement(obj) {
```
Comment (Ikilonchic) [inline src/AuthPM.js]: Add a check if the element is in the current DOM.

**Review by Ikilonchic — CHANGES_REQUESTED** [review]

File: src/AuthPM.js (hunk 1 above)
Comment (alexgul1) [reply src/AuthPM.js]: fixed in 375899d193f56ac48c9856603b5171ecb7f8d412

**Review by matroskin062 — APPROVED** [review]

File: src/AuthPM.js [hunk 2]
```diff
@@ -1,6 +1,86 @@
...
+
+  constructor(appId, target) {
+    this.#appId = appId;
+
+    this.#setTarget(target);
+
+    this.#target && this.#getAppList();
+  }
+
+  #setTarget(target) {
```
Comment (Ikilonchic) [inline src/AuthPM.js:16]: Maybe it would be better to write target setter here (es6).

File: src/AuthPM.js [hunk 3]
```diff
@@ -1,6 +1,86 @@
...
+  #renderSocials(socials) {
+    socials.forEach(({ id, name }) => {
+      const btn = this.#renderButton({ id, name });
+
+      this.#target.appendChild(btn);
+    });
+  }
+
+  #renderButton({ id, name }) {
+    const clickHandler = (event) => {
```
Comment (Ikilonchic) [inline src/AuthPM.js:59]: We need to save the handler to remove it on unmount.

Comment (alexgul1) [reply src/AuthPM.js:59]: I think that this logic should be left to the user of the widget

**Review by Ikilonchic — APPROVED** [review]

**Pull request merged**